lib/execution_segment.go

```diff
@@ -394,13 +401,276 @@ func (ess ExecutionSegmentSequence) String() string {
 	return strings.Join(result, ",")
 }
 
-// GetStripedOffsets returns everything that you need in order to execute only
-// the iterations that belong to the supplied segment...
+// lowest common denominator
+// https://en.wikipedia.org/wiki/Least_common_multiple#Using_the_greatest_common_divisor
+func (ess ExecutionSegmentSequence) lcd() int64 {
+	var acc = ess[0].length.Denom().Int64()
+	var n int64
+	for _, seg := range ess[1:] {
+		n = seg.length.Denom().Int64()
+		if acc == n || acc%n == 0 { // short circuit
+			continue
+		}
+		acc *= (n / gcd(acc, n))
+	}
+
+	return acc
+}
+
+// Greatest common divisor
+// https://en.wikipedia.org/wiki/Euclidean_algorithm
+func gcd(a, b int64) int64 {
+	for a != b {
+		if a > b {
+			a -= b
+		} else {
+			b -= a
+		}
+	}
+	return a
+}
+
+type sortInterfaceWrapper struct { // TODO: rename ? delete ? and replace ?
```

Comment:
I think we can call this "good enough for now", leave the `TODO`s and refactor it at some later point. We can figure out better names, code structure, and maybe further optimizations once everything else is working, so please add a new `low prio` + `new executors` + `refactor` issue for this.

Reply:
https://github.com/loadimpact/k6/issues/1379

I wrote it about the ExecutionTuple as sortInterfaceWrapper, in reality, will probably get removed when we next refactor it and lib.ExecutionTuple is a public type :D